python/ray/autoscaler/local/node_provider.py

```
@@ -90,12 +111,75 @@ class LocalNodeProvider(NodeProvider):
+            provider_config, cluster_name = self._get_node_configs(
+                provider_config, cluster_name)
+
         NodeProvider.__init__(self, provider_config, cluster_name)
-        self.state = ClusterState("/tmp/cluster-{}.lock".format(cluster_name),
-                                  "/tmp/cluster-{}.state".format(cluster_name),
-                                  provider_config)
+        self.state = ClusterState(
+            "/tmp/cluster-{}.lock".format(cluster_name),
+            "/tmp/cluster-{}.state".format(cluster_name),
+            provider_config,
+        )
+
+    def _get_node_configs(self, provider_config, cluster_name):
+        # Concatenate user_id to make cluster names unique between users.
+        new_cluster_name = cluster_name + "_" + getpass.getuser()
+        request = {
+            "request_type": "get_node_ips",
+            "cluster_name": new_cluster_name,
+            "num_workers": provider_config["max_workers"],
+        }
+        node_ips = self.get_http_response(request, self.server_address)
+        if "head_ip" not in node_ips:
+            raise OSError("Not enough nodes are available." +
+                          " Consider requesting less workers.")
+        else:
+            provider_config["head_ip"] = node_ips["head_ip"]
+            provider_config["worker_ips"] = node_ips["worker_ips"]
+        return provider_config, new_cluster_name
+
+    @staticmethod
+    def get_http_response(request, server_address):
+        headers = {
+            "Content-Type": "application/json",
+        }
+        request_message = json.dumps(request).encode()
+        http_server_address = "http://" + server_address
+        try:
+            r = requests.get(
+                http_server_address,
+                data=request_message,
+                headers=headers,
+                timeout=None,
+            )
+        except (RemoteDisconnected, ConnectionError):
+            logger.error(
```

logger.exception() -- do not hide the exception.